src/external/s2/util/math/mathutil.h

```diff
@@ -379,7 +381,7 @@ class MathUtil {
     // is no advantage to passing an argument type of "float" on Intel
     // architectures anyway.
 
-#if defined __GNUC__ && (defined __i386__ || defined __SSE2__)
+#if defined __GNUC__ && (defined __i386__ || defined __SSE2__) && !(REALM_WATCHOS && !REALM_APPLE_DEVICE)
```

Comment:
I'm a bit wary infesting s2 code with realm specific things. Are you sure we want to buildin the whole feature for watchos and appletv be default? I can hardly imagine this being used in this devices. The are either very tiny or specialised. Is there the real need for geo queries on watches?

Reply:
I lean towards core being agnostic about making decisions on which platforms to build for. It is unlikely that users will want to use geospatial on watchos, but we can leave that decision to the SDK.